google/cloud/storage/internal/grpc_hmac_key_metadata_parser_test.cc

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/internal/grpc_hmac_key_metadata_parser.h"
+#include "google/cloud/storage/internal/hmac_key_metadata_parser.h"
+#include "google/cloud/testing_util/is_proto_equal.h"
+#include <google/protobuf/text_format.h>
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace internal {
+namespace {
+
+namespace storage_proto = ::google::storage::v2;
+using ::google::cloud::testing_util::IsProtoEqual;
+using ::google::protobuf::TextFormat;
+
+TEST(GrpcHmacKeyMetadataParser, Roundtrip) {
+  storage_proto::HmacKeyMetadata input;
+  // To get the dates in RFC-3339 format I used:
+  //     date --rfc-3339=seconds --date=@1652099696  # Create
+  //     date --rfc-3339=seconds --date=@1652186096  # Update
```

Comment:
It doesn't look like we use a 3339 format anywhere here?

Reply:
Doh! Fixed (see below).